docs/README.md

```diff
@@ -0,0 +1,263 @@
+Configurations and variables can become lengthy and unwieldy, as a general note of advice it is best to move variables into a `group_vars` folder.
+```
+./inventory
+├── Cluser_A
+│   ├── group_vars
+│   │   ├── all.yml
+│   │   ├── rke2_agents.yml
+│   │   └── rke2_servers.yml
+│   └── hosts.yml
+└── Cluser_B
+    ├── group_vars
+    │   ├── all.yml
+    │   ├── rke2_agents.yml
+    │   └── rke2_servers.yml
+    └── hosts.yml
+
+5 directories, 8 files
+```
+
+
+## Enabling SELinux  
+Enabling SELinux in the playbook requires `selinux: true` be set in either the cluster, group, or host level config profiles (Please see [Special Variables](#special-variables) for more info). Though generally this should be set at the cluster and can be done like so:  
+__hosts.yml:__  
+```yaml
+---
+all:
+  vars:
+    cluster_rke2_config:
+      selinux: true
+```
+For more information please see the RKE2 documentation, [here](https://docs.rke2.io/security/selinux).
+
+
+## Enabling CIS Modes  
+Enabling the CIS tasks in the playbook requires a CIS profile be added to the ansible variables file. This can be placed in either the cluster, or group level config profiles (Please see [Special Variables](#special-variables) for more info). Below is an example, in the example the CIS profile is set at the group level, this ensures all server nodes run the CIS hardening profile tasks.  
+__hosts.yml:__  
+```yaml
+rke2_cluster:
+  children:
+    rke2_servers:
+      vars:
+        group_rke2_config:
+          profile: cis 
+```
+For more information please see the RKE2 documentation, [here](https://docs.rke2.io/security/hardening_guide).
+
+
+## Special Variables  
+In general this repository has attempted to move away from special or "magic" variables, however some are unavoidable, the (non-exhaustive) list of variables is below:    
+  - `all.vars.rke2_install_version`: This defines what version of RKE2 to install 
+  - `rke2_cluster.children.rke2_servers.vars.hosts.<host>.node_labels`: Defines a list of node labels for a specific server node     
+  - `rke2_cluster.children.rke2_agents.vars.hosts.<host>.node_labels`: Defines a list of node labels for a specific agent node  
+
+
+### RKE2 Config Variables  
+There are three levels an RKE2 config variables can be placed in, that is `cluster_rke2_config`, `group_rke2_config`, and `host_rke2_config`. 
+  - `all.vars.cluster_rke2_config`: Defines common RKE2 config options for the whole cluster
+  - `rke2_cluster.children.rke2_servers.vars.group_rke2_config`: Defines common RKE2 config options for the `rke2_servers` group  
+  - `rke2_cluster.children.rke2_agents.vars.group_rke2_config`: Defines common RKE2 config options for the `rke2_agents` group   
+  - `rke2_cluster.children.rke2_servers.vars.hosts.<host>.host_rke2_config`: Defines a list of node labels for a specific agent node    
+  - `rke2_cluster.children.rke2_agents.vars.hosts.<host>.host_rke2_config`: Defines a list of node labels for a specific agent node   
```

Comment:
Defines a list of node labels for a specific agent node
should be
Defines RKE2 config options for a specific agent node